aea/helpers/dependency_tree.py

```diff
@@ -187,20 +189,63 @@ def flatten_tree(
             flat_tree[level].append(package)
             cls.flatten_tree(dependencies, flat_tree, level + 1)
 
+    @staticmethod
+    def find_packages_in_a_project(
+        project_dir: Path,
+    ) -> List[Tuple[str, Path]]:
+        """Find packages in an AEA project."""
+
+        packages = []
+        for package_type, config_file in COMPONENTS:
+            _packages = [
+                *Path(project_dir, VENDOR).glob(f"*/{package_type}s/*/{config_file}"),
+                *Path(project_dir, f"{package_type}s").glob(f"*/{config_file}"),
```

Comment:
So I've added more strict glob patterns, think this should address your concerns @DavidMinarsch